src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/hotshard/HotShardClusterRca.java

```diff
@@ -52,7 +52,8 @@ public class HotShardClusterRca extends Rca<ResourceFlowUnit<HotClusterSummary>>
     private Set<String> unhealthyNodes;
 
     // Guava Table with Row: 'Index_Name', Column: 'NodeShardKey', Cell Value: 'Value'
-    // TODO: Use the fact that we're getting at max topK*2 consumers from each node and perform further optimization.
+    // TODO: Use the fact that we're getting at max topK*2 consumers from each node and perform
+    // further optimization.
```

Comment:
Instead of passing `topK*2` flow units per node, can we make this `topK`. This would require updating the HotShardSummary to contain `cpu_utilization`, `cpu_utilization_threshold`, `heap_alloc_rate`, `heap_alloc_rate_threshold`, `criteria`, where the `criteria` can be `cpu_utilization` and/or `heap_alloc_rate`

What do you think ?

Reply:
Done. I think that works well as statistical properties of the metrics show that in majority of cases we have both criteria matched, that way utilizing the network and heap a lot better. I introduced a CriteriaEnum for this. Also removed the thresholds as they were never used nor persisted later in `HotShardClusterRCA` and just consumed resources. Please raise additional concerns if any.